contracts/src/v0.8/ccip/libraries/ERC165CheckerReverting.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import {IERC165} from "../../vendor/openzeppelin-solidity/v4.8.3/contracts/utils/introspection/IERC165.sol";
+
+/**
+ * @dev Library used to query support of an interface declared via {IERC165}.
+ *
+ * Note that these functions return the actual result of the query: they do not
+ * `revert` if an interface is not supported. It is up to the caller to decide
+ * what to do in these cases.
+ *
+ * Note this is exactly the same as the OZ version, with the exception that external calls will revert
+ * if < 31_000 gas is available, to prevent message delivery issues in CCIP.
+ */
+library ERC165CheckerReverting {
+  // As per the EIP-165 spec, no interface should ever match 0xffffffff
+  bytes4 private constant _INTERFACE_ID_INVALID = 0xffffffff;
+
+  // bytes4(keccak256("NotEnoughGasForSupportsInterfaceCall()"))
```

Comment:
probably good to also define the error so we can reference it in tests and check it's actually the same

Reply:
I'll just replace it with the actual error and then use top-level solidity to insert it into the assembly. Much easier.